### PR TITLE
Fix incorrect JsonProvider numeric inference

### PR DESCRIPTION
**Pull request by panesofglass**
See #1221 

Hypothesis is that commit 8d05e4bf32fe750cb0f1bad977cbd5b97b39ab17 is not quite correct.

### CODE REVIEW

File: tests/FSharp.Data.Tests/JsonProvider.fs [hunk 1]
```diff
@@ -734,3 +734,18 @@ let ``Getting a large float at runtime when an integer was inferred should throw
     let f = 1234567890123456789012345678901234567890.
     let json = JsonProvider<"""{ "x" : 0.500, "y" : 0.000 }""">.Parse("""{ "x" : -0.250, "y" : 1234567890123456789012345678901234567890 }""")
     (fun () -> json.Y) |> shouldThrow (sprintf "Expecting an Int32 at '/y', got %s" (f.ToString("E14").Replace("+0", "+").Replace(",", ".")))
+
+[<Test>]
+let ``Getting field with numeric values should infer decimal``() =
+    let json = JsonProvider<""" { "a":[1.0, 1.0] } """>.GetSample()
```
Comment (panesofglass) [inline tests/FSharp.Data.Tests/JsonProvider.fs]: I _think_ this is supposed to infer `decimal`, but I could be wrong.

**Comment by panesofglass** [comment]
Appears these lines are the issue: https://github.com/fsharp/FSharp.Data/blob/8d05e4bf32fe750cb0f1bad977cbd5b97b39ab17/src/Json/JsonInference.fs#L34-L35

What's wanted here? I would expect the format of the number to override the use of `Math.Round(v) = v` as a means of inference. If not, what is required to make this work correctly?

**Comment by panesofglass** [comment]
@ovatsus @baronfel I don't know the right way to fix this. Help is appreciated.

**Comment by baronfel** [comment]
Personally I think the single flag for `inferTypesFromValues` is a bit wide for coercing both strings _and_ numeric types.  I'd be in favor of expanding the number of static/configuration parameters for this (maybe just a split for string-inference and number-inference) as well as expanding the number of JsonValue cases to include `Integer` and `LongInteger` or something semantically similar, since the only time where we have enough information is at JsonValue.Parse time. Armed with JsonValue.Integer and JsonValue.LongInteger we could easily handle type mapping to int/int64 for json provider even without `inferTypesFromValues` set.

**Comment by dsyme** [comment]
Close/open to re-run CI

**Comment by dsyme** [comment]
Closing this out as the repo has diverged a long way since the proposal was made

**Comment by panesofglass** [comment]
We ultimately never upgraded and are stuck on FSharp.Data 3.3.3.

**Pull request closed without merging**